### PR TITLE
[TieredStorage] Deprecate the use of account-hash in HotStorage

### PR DESCRIPTION
**Pull request by yhchiang-sol**
#### Problem
TieredStorage stores account hash as an optional field inside its HotStorage.
However, the field isn't used and we have already decided to deprecate
the account hash.

#### Summary of Changes
Remove account-hash from the tiered-storage.

#### Test Plan
Existing tiered-storage tests.
Running validators w/ tiered-storage in mainnet-beta w/o storing account-hash.

### CODE REVIEW

File: accounts-db/src/tiered_storage/meta.rs [hunk 1]
```diff
@@ -102,20 +95,15 @@ impl AccountMetaFlags {
 /// Note that the storage representation of the optional fields might be
 /// different from its in-memory representation.
 #[derive(Debug, PartialEq, Eq, Clone)]
-pub struct AccountMetaOptionalFields<'a> {
```
Comment (brooksprumo) [inline accounts-db/src/tiered_storage/meta.rs:105]: Removing the lifetime is a nice bonus!

File: accounts-db/src/tiered_storage/readable.rs [hunk 2]
```diff
@@ -42,7 +42,7 @@ impl<'accounts_file, M: TieredAccountMeta> TieredReadableAccount<'accounts_file,
 
     /// Returns the hash of this account.
     pub fn hash(&self) -> Option<&'accounts_file AccountHash> {
-        self.meta.account_hash(self.account_block)
+        None
     }
```
Comment (brooksprumo) [inline accounts-db/src/tiered_storage/readable.rs]: I think we should remove this function entirely.

Comment (yhchiang-sol) [reply accounts-db/src/tiered_storage/readable.rs]: Removed, and also improved the comment from its StoredAccountMeta call site:

```
    pub fn hash(&self) -> &'storage AccountHash {
        match self {
            Self::AppendVec(av) => av.hash(),
            // tiered-storage has deprecated the use of AccountHash
            Self::Hot(_) => &DEFAULT_ACCOUNT_HASH,
        }
    }
```

**Review by brooksprumo — APPROVED** [review]
:shipit:

**Comment by jeffwashington** [comment]
For compressed state, we need to store the hash per account. We also likely need to store a heterogenous mixture of full accounts or hash.
It seems it is costing us 1 bit per account to be able to store an account's hash.
My inclination is that this pr should be reverted. Thoughts?

**Comment by yhchiang-sol** [comment]
> For compressed state, we need to store the hash per account. We also likely need to store a heterogenous mixture of full accounts or hash.
It seems it is costing us 1 bit per account to be able to store an account's hash.
My inclination is that this pr should be reverted. Thoughts?

If we want to use the minimum storage size to handle state-compressed accounts, then having a dedicated block for all the state-compressed accounts would probably lead to the best result.  The resulting tiered-storage file would look something like this:

* accounts blocks
* index block
* owners block
* state compressed block
* filter block (e.g. a bloomfilter to determine whether an account might exist in this file)
* footer

A state-compressed block can be simply:

```
<pubkey-1><hash-1>
<pubkey-2><hash-2>
...
```

**Pull request merged**